cairo/ethereum/cancun/vm.cairo

```diff
@@ -88,6 +105,311 @@ struct MessageStruct {
+        cast(accounts_to_delete_end, DictAccess*),
+    );
+    tempvar new_accounts_to_delete = SetAddress(
+        new SetAddressStruct(
+            dict_ptr_start=cast(accounts_to_delete_start, SetAddressDictAccess*),
+            dict_ptr=cast(new_accounts_to_delete_end, SetAddressDictAccess*),
+        ),
+    );
+
+    // Squash & update accessed_addresses into parent
+    let accessed_addresses = evm.value.accessed_addresses;
+    let accessed_addresses_start = accessed_addresses.value.dict_ptr_start;
+    let accessed_addresses_end = accessed_addresses.value.dict_ptr;
+    let new_accessed_addresses_end = squash_and_update(
+        cast(child_evm.value.accessed_addresses.value.dict_ptr_start, DictAccess*),
+        cast(child_evm.value.accessed_addresses.value.dict_ptr, DictAccess*),
+        cast(accessed_addresses_end, DictAccess*),
+    );
+
+    tempvar new_accessed_addresses = SetAddress(
+        new SetAddressStruct(
+            dict_ptr_start=cast(accessed_addresses_start, SetAddressDictAccess*),
+            dict_ptr=cast(new_accessed_addresses_end, SetAddressDictAccess*),
+        ),
+    );
+
+    // Squash & update accessed_storage_keys into parent
+    let accessed_storage_keys = evm.value.accessed_storage_keys;
+    let accessed_storage_keys_start = accessed_storage_keys.value.dict_ptr_start;
+    let accessed_storage_keys_end = accessed_storage_keys.value.dict_ptr;
+    let new_accessed_storage_keys_end = squash_and_update(
+        cast(child_evm.value.accessed_storage_keys.value.dict_ptr_start, DictAccess*),
+        cast(child_evm.value.accessed_storage_keys.value.dict_ptr, DictAccess*),
+        cast(accessed_storage_keys_end, DictAccess*),
+    );
+
+    tempvar new_accessed_storage_keys = SetTupleAddressBytes32(
+        new SetTupleAddressBytes32Struct(
+            dict_ptr_start=cast(accessed_storage_keys_start, SetTupleAddressBytes32DictAccess*),
+            dict_ptr=cast(new_accessed_storage_keys_end, SetTupleAddressBytes32DictAccess*),
+        ),
+    );
+
+    // Squash dropped dicts
+    dict_squash(
+        cast(child_evm.value.stack.value.dict_ptr_start, DictAccess*),
+        cast(child_evm.value.stack.value.dict_ptr, DictAccess*),
+    );
+
+    dict_squash(
+        cast(child_evm.value.memory.value.dict_ptr_start, DictAccess*),
+        cast(child_evm.value.memory.value.dict_ptr, DictAccess*),
+    );
+
+    dict_squash(
+        cast(child_evm.value.valid_jump_destinations.value.dict_ptr_start, DictAccess*),
+        cast(child_evm.value.valid_jump_destinations.value.dict_ptr, DictAccess*),
+    );
+
+    // No need to squash the message's `accessed_addresses` and `accessed_storage_keys` because
+    // they were moved into the Evm, which we just squashed.
+
+    // No need to squash the env's `state` and `transient_storage` because it was handled in `rollback_transaction`,
+    // when we squashed and appended the prev keys to the parent's state and transient storage segments.
```

Comment:
What about the child_evm message `accessed_addresses` and `accessed_storage_keys`?
Also why rollback_transaction is mentioned even though it is the success case? Shouldn't this be commit_transaction?

Reply:
either rollback or commit, yes, `close_transaction`